### ISSUE TITLE
tabular_pipeline(TabICLClassifier()) ?

### ISSUE THREAD

**Issue by jeromedockes** [skrub-data/skrub#2150]
### Problem Description

the default `tabular_pipeline('classifier')` gives a fairly fast, tough-to-beat baseline that *uses only required dependencies of skrub*. I wonder if it would make sense to also provide a configuration that lifts that constraint to get a possibly even stronger model:

- using the TextEncoder rather than StringEncoder, or making that choice a hyperparameter (requires installing sentence-transformers, slower than StringEncoder)
- using TabICL instead of the histogram gradient boosting (requires installing TabICL)

if we don't want to go as far as tabular_pipeline('tabicl_classifier') we may still special-case TabICL models (the same way we do hgb and random forest) to pick the best preprocessing for them; it would be interesting to hear what @dholzmueller thinks what are the best TableVectorizer parameters (and other preprocessing steps) to use in that case

**Comment by dholzmueller** [comment]
That would be nice! Re TableVectorizer parameters:
- low-cardinality should be "passthrough" so TabICL can use its own categorical handling
- probably more (periodic) datetime features would help
- In STRABLE (https://arxiv.org/pdf/2605.12292) StringEncoder is actually better than LLMs for TabICLv2, but this might differ when it's only used for high cardinality, so I don't know what would work best here. But we could leave it at the StringEncoder for a fast default.
- I don't know what's the best cardinality threshold, TabICL has only seen up to cardinalities 9 or 10 during pre-training. With the StringEncoder maybe that would be a good threshold, but we could also leave it at 40 to make it faster.
- The rest should be fine (TabICL does its own imputation and numerical preprocessing).

**Comment by jeromedockes** [comment]
thanks for that info! I guess stringencoder vs textencoder depends more on the content of the column so without more information or the ability to tune it stringencoder is possibly a safer bet indeed

**Comment by rcap107** [comment]
Discussed this IRL. 

Good to have, but we shouldn't add TabICL to the optional dependencies. We can still modify the tabular_pipeline so that if it detects a TabICL estimator it will select the best parameters for the TableVectorizer as explained above. 

There should also be an example about this, maybe we can add it when we address #2144.

**Comment by jeromedockes** [comment]
we can use something like 
```
estimator.__module__.split('.')[0] == 'tabicl' and estimator.__class__.__name__ in ['TabICLClassifier', 'TabICLRegressor']
```

to decide if we should use the preprocessing params that are most adequate for tabicl

or maybe just `estimator.__class__.__name__ in ['TabICLClassifier', 'TabICLRegressor']` is probably enough

**Comment by ashwinvis** [comment]
@jeromedockes looking into it.

**Comment by jeromedockes** [comment]
https://tabicl.readthedocs.io/en/latest/tutorials/string_handling.html

```diff
diff --git a/skrub/_tabular_pipeline.py b/skrub/_tabular_pipeline.py
index ac1c0124..9f500fbb 100644
--- a/skrub/_tabular_pipeline.py
+++ b/skrub/_tabular_pipeline.py
@@ -268,6 +268,12 @@ def tabular_pipeline(estimator, *, n_jobs=None):
             ),
             high_cardinality=StringEncoder(),
         )
+    elif estimator.__class__.__name__.startswith("TabICL"):
+        vectorizer.set_params(
+            low_cardinality="passthrough",
+            cardinality_threshold=10,
+           datetime=DatetimeEncoder(periodic_encoding="spline")
+            )
     else:
         vectorizer.set_params(datetime=DatetimeEncoder(periodic_encoding="spline"))
     steps = [vectorizer]
```
